hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java

```diff
@@ -171,6 +171,11 @@
   /** Configuration key for master web API port */
   public static final String MASTER_INFO_PORT = "hbase.master.info.port";
 
+  /** Configuration key for the list of master host:ports **/
+  public static final String MASTER_ADDRS_KEY = "hbase.master.addrs";
```

Comment:
Make sure parsing can handle both <host> and <host>:<port> comma separated values. The other configuration setting like this is the zk quorum, which accepts a comma-separated list of <host>:<port> or <host> . Let's keep things consistent.

Reply:
Just to confirm my understanding is correct, you mean the port number is optional right ? Something like "foo.com,bar.com:1234,baz.com:5678,temp.com"

Reply:
Looking at our zookeeper connection string parsing code. We depend on the Hadoop `Configuration` object to handle parsing of the comma-delimited configuration value. We then manually check for `':'` characters to split out ports.

If we're making a habit of doing this kind of parsing (lists of socket addresses), it's probably work encapsulating the logic into a single place.

https://github.com/apache/hbase/blob/5e84997f2ffdbcf5f849d70c30ddbe2db4039ca4/hbase-common/src/main/java/org/apache/hadoop/hbase/zookeeper/ZKConfig.java#L97-L111

Reply:
Yep, I'll consolidate the parsing logic. It will come in the next patch.

Reply:
You plan to address this last nit? I'm +1 with this bit handled.

Reply:
@ndimiduk The parsing logic will come as a part of PR for HBASE-23305. There is nothing in this patch that consumes the content of "hbase.master.addrs".